cloud/openstack/os_network.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+
+# Copyright (c) 2014 Hewlett-Packard Development Company, L.P.
+# Copyright (c) 2013, Benno Joy <benno@ansible.com>
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_network
+short_description: Creates/Removes networks from OpenStack
+extends_documentation_fragment: openstack
+version_added: "1.10"
+description:
+   - Add or Remove network from OpenStack.
+options:
+   state:
+     description:
+        - Indicate desired state of the resource.
+     choices: ['present', 'absent']
+     default: present
+   name:
+     description:
+        - Name to be assigned to the network.
+     required: true
+     default: None
+   shared:
+     description:
+        - Whether this network is shared or not.
+     required: false
+     default: false
+   admin_state_up:
+     description:
+        - Whether the state should be marked as up or down.
+     required: false
+     default: true
+requirements: ["shade"]
+'''
+
+EXAMPLES = '''
+- os_network:
+    name=t1network
+    state=present
+'''
+
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=True),
+        shared=dict(default=False, type='bool'),
+        admin_state_up=dict(default=True, type='bool'),
+    )
+
+    module_kwargs = openstack_module_kwargs()
+    module = AnsibleModule(argument_spec, **module_kwargs)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    state = module.params['state']
+    name = module.params['name']
+    shared = module.params['shared']
+    admin_state_up = module.params['admin_state_up']
+
+    try:
```

Comment:
exception handling is a bit too 'catchall' it would be nice to capture and return more meaningful messages from module 'failted to connect: <exception reason>', 'failed to create network', etc.